doc/Protocols/pr-mrest_x-road_service_metadata_protocol_for_rest.md

```diff
@@ -0,0 +1,239 @@
+
+See X-Road terms and abbreviations documentation \[[TA-TERMS](#Ref_TERMS)\].
+
+### 1.2 References
+
+1. <a name="Ref_PR-META" class="anchor"></a>\[PR-META\] X-Road: Service Metadata Protocol. Document ID:
+[PR-META](pr-meta_x-road_service_metadata_protocol.md).
+2. <a name="Ref_RFC2119" class="anchor"></a>\[RFC2119\] Key words for use in RFCs to Indicate Requirement Levels, Internet Engineering Task Force, 1997,
+[https://www.ietf.org/rfc/rfc2119.txt](https://www.ietf.org/rfc/rfc2119.txt)
+3. <a id="Ref_TERMS" class="anchor"></a>\[TA-TERMS\] X-Road Terms and Abbreviations. Document ID: [TA-TERMS](../terms_x-road_docs.md).
+4. <a id="Ref_PR-REST" class="anchor"></a>\[PR-REST\] X-Road: Message Protocol for REST. Document ID:
+[PR-REST](pr-rest_x-road_message_protocol_for_rest.md).
+
+## 2 Retrieving List of Service Providers
+
+For retrieving the list of service providers listClients metaservice is used. It can be invoked with simple HTTP GET to right URL. Service output format is controlled with Accept header. The details of listClients are described in \[[PR-META](#Ref_PR-META)\].
+
+## 3 Retrieving List of Central Services
+
+For retrieving the list of central services listCentralServices metaservice is used. It can be invoked with simple HTTP GET to right URL. The details of listCentralServices are described in \[[PR-META](#Ref_PR-META)\].
+
+## 4 Retrieving List of Services
+
+X-Road provides two methods for getting the list of services offered by an X-Road client:
+
+* `listMethods` lists all REST services offered by a service provider.
+
+* `allowedMethods` lists all REST services offered by a service provider that the caller has permission to invoke.
+
+Both methods are invoked as regular X-Road REST services (see specification \[[PR-REST](#Ref_PR-REST)\] for details on the X-Road REST protocol).
+
+The serviceId MUST contain the identifier of the target service provider and the value of the serviceCode element MUST be either `listMethods` or `allowedMethods`.
+
+Request example
+```
+GET /r1/INSTANCE/CLASS2/MEMBER2/SUBSYSTEM2/listMethods
+```
+HTTP request headers
+```
+X-Road-Client: INSTANCE/CLASS1/MEMBER1/SUBSYSTEM1
+```
+
+The body of the response message MUST contain a list of services provided by the service provider (in case of listMethods) or open to the given client (in case of allowedMethods). The response SHALL NOT contain names of the metainfo services.
+
+Annex [A](#annex-a-service-descriptions-for-rest-metadata-services) contains the OpenAPI description of the REST metadata services.
+
+Annexes [B.1](#c1-listmethods-response) and [B.2](#c2-allowedmethods-response) contain example response messages for services, respectively.
+
+## 5 Retrieving the OpenAPI description of a Service
+
+TBD
+
+## Annex A Service Descriptions for REST Metadata Services
+
+```yaml
+openapi: 3.0.0
+info:
+  title: X-Road Service Metadata API for REST
+  version: '0.1'
+servers:
+  - url: 'https://{securityserver}/'
+paths:
+  /listMethods:
```

Comment:
the path should probably be `/{instanceId}/{memberClass}/{memberCode}/{subsystemCode}/listMethods`
(`GET https://securityserver/listMethods` request won't work). Allowedmethods similarly.